### PR TITLE
fix(treeview): Adjust default TreeViewItem template to include ContentTemplateSelector

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno/issues/11942

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

Adjusts the default `TreeViewItem` template to properly template bind the `ContentTemplateSelector` property.

Related to https://github.com/unoplatform/uno/issues/3519 and https://github.com/unoplatform/uno/issues/857

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
I think there are other occurrences of this workaround in the styles - if it is Uno specific, I would suggest including a XML comment with link to the issue 👍

File: src/Uno.UI.FluentTheme.v2/Resources/Version2/PriorityDefault/TreeViewItem.xaml
Comment (MartinZikmund) [inline src/Uno.UI.FluentTheme.v2/Resources/Version2/PriorityDefault/TreeViewItem.xaml:1]: I think the update to the merged XAMLs should be committed too (e.g. For V2 https://github.com/unoplatform/uno/blob/fcc93dc93995b0641d6c3b1b1f20480232bcfaee/src/Uno.UI.FluentTheme.v2/themeresources_v2.xaml)

Comment (jeromelaban) [reply src/Uno.UI.FluentTheme.v2/Resources/Version2/PriorityDefault/TreeViewItem.xaml:1]: Indeed.

**Comment by jeromelaban** [comment]
The fix is invalid and needs further ContentControl/ContentPresenter alignment.

**Pull request closed without merging**